source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -2050,6 +2114,39 @@ However,
 if the server still doesn't respond when the monitor attempts to reconnect,
 then it is probably down.
 
+Use error messages to detect "not master" and "node is recovering"
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+When error codes are not available, error messages are checked for the
+substrings "not master" and "node is recovering". This is because driver
+authors have found the substrings to be **more** stable than error codes.
```

Comment:
Correction, "This is because older server versions returned unstable error codes or no error codes in many circumstances."

Reply:
Done.